src/app/calculator/utilities/power-factor-triangle/power-factor-triangle-form/power-factor-triangle-form.component.ts

```diff
@@ -36,7 +36,7 @@ export class PowerFactorTriangleFormComponent implements OnInit {
   }
 
   calculate() {
-    if (this.data.apparentPower != null && this.data.realPower != null && this.data.reactivePower != null && this.data.phaseAngle != null && this.data.powerFactor != null) {
+    if (this.data.apparentPower != null && this.data.realPower != null && this.data.reactivePower != null && (this.data.phaseAngle != null && this.data.phaseAngle <= 90) && (this.data.powerFactor != null && (this.data.powerFactor <= 1 && this.data.powerFactor > 0))) {
```

Comment:
This could be quickly cleaned up. Can we ignore the null values and use convertNullInputValueForObjectConstructor() from the suite helper service if the suite is complaining? Otherwise set the not null chain and real validation logical statements as two variables.